src/modules/menu/components/menu/MenuTray.vue

```diff
@@ -1,10 +1,102 @@
+import MenuAdvancedToolsList from '@/modules/menu/components/advancedTools/MenuAdvancedToolsList.vue'
+import MenuSection from '@/modules/menu/components/menu/MenuSection.vue'
+import MenuPrintSection from '@/modules/menu/components/print/MenuPrintSection.vue'
+import MenuSettings from '@/modules/menu/components/settings/MenuSettings.vue'
+import MenuShareSection from '@/modules/menu/components/share/MenuShareSection.vue'
+import MenuTopicSection from '@/modules/menu/components/topics/MenuTopicSection.vue'
+
+const dispatcher = { dispatcher: 'MenuTray.vue' }
+
+const i18n = useI18n()
+const store = useStore()
+
+const props = defineProps({
+    compact: {
+        type: Boolean,
+        default: false,
+    },
+})
+const { compact } = toRefs(props)
+
+const refs = ref({})
+// multiMenuSections means that they can be open together
+const multiMenuSections = ref(['topicsSection', 'activeLayersSection'])
+// singleModeSections means that those section cannot be open together with other
+// sections and would therefore toggle other sections automatically.
+const singleModeSections = ref([
+    'drawSection',
+    'settingsSection',
+    'shareSection',
+    'toolsSection',
+    'printSection',
+    '3dSection',
+])
+
+const is3dMode = computed(() => store.state.cesium.active)
+const showImportFile = computed(() => store.state.ui.importFile)
+const showDrawingOverlay = computed(() => store.state.drawing.drawingOverlay.show)
+const mapModuleReady = computed(() => store.state.app.isMapReady)
+
+watch(showImportFile, (show) => {
+    if (show) {
+        refs.value.activeLayersSection.open()
+    }
+})
+
+onMounted(() => {
+    if (is3dMode.value) {
+        refs.value['3dSection'].open()
+    }
+})
+
+function toggleDrawingOverlay() {
+    store.dispatch('toggleDrawingOverlay', dispatcher)
+}
+
+function onOpenMenuSection(id) {
+    let toClose = singleModeSections.value.filter((section) => section !== id)
+    if (singleModeSections.value.includes(id)) {
+        toClose = toClose.concat(multiMenuSections.value)
+    }
+    toClose.forEach((section) => refs.value[section]?.close())
+}
+
+function onCloseMenuSection(id) {
+    if (['drawSection', 'toolsSection'].includes(id)) {
+        refs.value.activeLayersSection.open()
+    }
+}
+
+/**
+ * Add the element reference to the refs object. The reference can be then retrieved by it's id
+ * attribute
+ *
+ * NOTE: To work the element requires an id attribute. For Component, it requires to expose an "id"
+ * constant.
+ *
+ * @param {any} el Reference to the element
+ */
+function addRefById(el) {
+    if (el !== null && !Object.keys(refs.value).includes(el.id)) {
+        refs.value[el.id] = el
+    }
+}
+</script>
+
 <template>
     <div data-cy="menu-tray-inner" :class="[{ 'menu-tray-compact': compact }, 'menu-tray-inner']">
         <MenuSection
             id="settingsSection"
-            ref="settingsSection"
+            :ref="addRefById"
```

Comment:
it would also be possible to directly give the correct "ref" name here (`multiMenuSections` or the single equivalent), having the same ref on multiple element makes them available as an array of ref in the `setup`

Reply:
But then you have an array and not an object and close/open a specific section you will need to do a search in the array by id.